packages/core/src/test/testUtil.ts

```diff
@@ -55,6 +64,59 @@ export function getWorkspaceFolder(dir: string): vscode.WorkspaceFolder {
 }
 
 /**
+ * An all-in-one solution for a folder to use during tests:
+ * - Cleans up after all tests complete
+ * - Each instance is isolated from other tests
+ * - Convenient methods for files operations on paths RELATIVE to the folder
+ *
+ * Create an instance using {@link TestFolder.create()}.
+ *
+ * Add any new methods to this class when needed.
+ */
+export class TestFolder {
```

Comment:
All new test files that need a temporary folder for testing should use this class. We should transition away from the existing methods over time, as indicated by the `deprecated` indicator in their docstrings.

Reply:
It looks like you're using nodeFs in the write function. Should we add a declaimer that parts of it only work in node?

Reply:
Right now this class is used in our agnostic FileSystem unit tests to do the setup. So this is why we dont use our agnostic FS in here YET, so we don't use our code to test itself. But if the day comes that we need it for web, we should be able to add some agnostic FS methods in here.

Adding this information to docstring